docs/plugins.rst

```diff
@@ -220,3 +220,26 @@ Note on role based views
 Airflow 1.10 introduced role based views using FlaskAppBuilder. You can configure which UI is used by setting
 rbac = True. To support plugin views and links for both versions of the UI and maintain backwards compatibility,
 the fields appbuilder_views and appbuilder_menu_items were added to the AirflowTestPlugin class.
+
+
+Plugins as Python packages
+--------------------------
+
+It is possible to load plugins via `setuptools' entrypoint<https://packaging.python.org/guides/creating-and-discovering-plugins/#using-package-metadata>`_ mechanism. To do this link
+your plugin using an entrypoint in your package. If the package is installed, airflow
+will automatically load the registered plugins from the entrypoint list.
+
+.. code-block:: python
+
+    from setuptools import setup
+
+    setup(
+        name="my-package",
+        ...
+        entry_points = {
+            'airflow.plugins': [
+                'my_plugin = my_package.my_plugin:MyAirflowPlugin'
+            ]
+        }
+    )
+
```

Comment:
I'll expand the documentation when I get home, but given a class like

```python
# my_package/my_plugin.py
from airflow.plugins_manager import AirflowPlugin

class MyAirflowPlugin(AirflowPlugin):
    name = 'my_namespace.MyPlugin'
```
I am quite sure (will test) that the plugin would be exposed under `airflow.operators.my_namespace.MyPlugin`.

Reply:
I'd write some test cases for this functionality, but it is a bit difficult given the structure of this file. Would there be objection to wrapping this functionality in some functions as I did with `is_valid_plugin(...)` to make this a bit easier to test?

Reply:
Before I spent time working on testcases, this would be the idea (and I'd put it in a separate PR) https://github.com/drewsonne/incubator-airflow/pull/1/files

Reply:
The class as you have it in that PR1 you link to looks fine from a very quick glance.

I was more thinking that in the setup.py example in the docs that the `my_plugin` (and maybe `MyAirflowPlugin`?) name doesn't matter - it's not part of the the "name" that is exposed by airflow is it? It would be `from airflow.operators.my_namespace import MyOperator` right?

Reply:
Ah right, i have not used the “name” of the entry point as it is already handled by the ‘AirflowPlugin.name’. We could use it, but it seems like a precedence would just complicate things.

Reply:
And yes, that import is correct